packages/components/src/unit-control/test/index.tsx

```diff
@@ -83,10 +86,14 @@ const ControlledSyncUnits = () => {
 	);
 };
 
+const ControlledUnitControl = ( { ...props }: UnitControlProps ) => {
+	return <UnitControl __next40pxDefaultSize { ...props } />;
+};
```

Comment:
To minimize the amount of changes in this file, I'm going to suggest something like this. The main points of the changes being:

- "Controlled"/"Uncontrolled" in these testing contexts refers to the [React concept](https://react.dev/reference/react-dom/components/input#controlling-an-input-with-a-state-variable), so naming this `ControlledUnitControl` would be confusing. But since we want the prop on all instances anyway, let's just import the original as `_UnitControl` and define an overridden `UnitControl` at the top.
- It's generally more accurate to use `React.ComponentProps<>` to extract types directly from the component, rather than import the `FooComponentProps`. In the component package, the actual component often allows more props than is defined in the `FooComponentProps` type.


```diff
diff --git a/packages/components/src/unit-control/test/index.tsx b/packages/components/src/unit-control/test/index.tsx
index b04d815cdb..ad98d57cae 100644
--- a/packages/components/src/unit-control/test/index.tsx
+++ b/packages/components/src/unit-control/test/index.tsx
@@ -12,9 +12,12 @@ import { useState } from '@wordpress/element';
 /**
  * Internal dependencies
  */
-import UnitControl from '..';
+import _UnitControl from '..';
 import { CSS_UNITS, parseQuantityAndUnitFromRawValue } from '../utils';
-import type { UnitControlProps } from '../types';
+
+const UnitControl = ( props: React.ComponentProps< typeof _UnitControl > ) => (
+	<_UnitControl __next40pxDefaultSize { ...props } />
+);
 
 const getInput = ( {
 	isInputTypeText = false,
@@ -71,13 +74,11 @@ const ControlledSyncUnits = () => {
 	return (
 		<>
 			<UnitControl
-				__next40pxDefaultSize
 				label="Field A"
 				value={ state.valueA }
 				onChange={ ( v ) => onUnitControlChange( 'valueA', v ) }
 			/>
 			<UnitControl
-				__next40pxDefaultSize
 				label="Field B"
 				value={ state.valueB }
 				onChange={ ( v ) => onUnitControlChange( 'valueB', v ) }
@@ -86,14 +87,10 @@ const ControlledSyncUnits = () => {
 	);
 };
 
-const ControlledUnitControl = ( { ...props }: UnitControlProps ) => {
-	return <UnitControl __next40pxDefaultSize { ...props } />;
-};
-
 describe( 'UnitControl', () => {
 	describe( 'Basic rendering', () => {
 		it( 'should render', () => {
-			render( <ControlledUnitControl /> );
+			render( <UnitControl /> );
 			const input = getInput();
 			const select = getSelect();
 
@@ -102,12 +99,10 @@ describe( 'UnitControl', () => {
 		} );
 
 		it( 'should render custom className', () => {
-			const { container: withoutClassName } = render(
-				<ControlledUnitControl />
-			);
+			const { container: withoutClassName } = render( <UnitControl /> );
 
 			const { container: withClassName } = render(
-				<ControlledUnitControl className="hello" />
+				<UnitControl className="hello" />
 			);
 
 			expect(
@@ -121,7 +116,7 @@ describe( 'UnitControl', () => {
 		} );
 
 		it( 'should not render select, if units are disabled', () => {
-			render( <ControlledUnitControl value="3em" units={ [] } /> );
+			render( <UnitControl value="3em" units={ [] } /> );
 			const input = getInput();
 			// Using `queryByRole` instead of `getSelect` because we need to test
 			// for this element NOT to be in the document.
@@ -132,11 +127,7 @@ describe( 'UnitControl', () => {
 		} );
 
 		it( 'should render label if single units', () => {
-			render(
-				<ControlledUnitControl
-					units={ [ { value: '%', label: '%' } ] }
-				/>
-			);
+			render( <UnitControl units={ [ { value: '%', label: '%' } ] } /> );
 
 			const select = screen.queryByRole( 'combobox' );
 			const label = screen.getByText( '%' );
@@ -151,9 +142,7 @@ describe( 'UnitControl', () => {
 			const user = userEvent.setup();
 			const onChangeSpy = jest.fn();
 
-			render(
-				<ControlledUnitControl value="50px" onChange={ onChangeSpy } />
-			);
+			render( <UnitControl value="50px" onChange={ onChangeSpy } /> );
 
 			const input = getInput();
 			await user.clear( input );
@@ -174,9 +163,7 @@ describe( 'UnitControl', () => {
 			const user = userEvent.setup();
 			const onChangeSpy = jest.fn();
 
-			render(
-				<ControlledUnitControl value="50px" onChange={ onChangeSpy } />
-			);
+			render( <UnitControl value="50px" onChange={ onChangeSpy } /> );
 
 			const input = getInput();
 			await user.type( input, '{ArrowUp}' );
@@ -192,9 +179,7 @@ describe( 'UnitControl', () => {
 			const user = userEvent.setup();
 			const onChangeSpy = jest.fn();
 
-			render(
-				<ControlledUnitControl value="50px" onChange={ onChangeSpy } />
-			);
+			render( <UnitControl value="50px" onChange={ onChangeSpy } /> );
 
 			const input = getInput();
 			await user.type( input, '{Shift>}{ArrowUp}{/Shift}' );
@@ -210,9 +195,7 @@ describe( 'UnitControl', () => {
 			const user = userEvent.setup();
 			const onChangeSpy = jest.fn();
 
-			render(
-				<ControlledUnitControl value={ 50 } onChange={ onChangeSpy } />
-			);
+			render( <UnitControl value={ 50 } onChange={ onChangeSpy } /> );
 
 			const input = getInput();
 			await user.type( input, '{ArrowDown}' );
@@ -228,9 +211,7 @@ describe( 'UnitControl', () => {
 			const user = userEvent.setup();
 			const onChangeSpy = jest.fn();
 
-			render(
-				<ControlledUnitControl value={ 50 } onChange={ onChangeSpy } />
-			);
+			render( <UnitControl value={ 50 } onChange={ onChangeSpy } /> );
 
 			const input = getInput();
 			await user.type( input, '{Shift>}{ArrowDown}{/Shift}' );
@@ -247,7 +228,7 @@ describe( 'UnitControl', () => {
 			const onChangeSpy = jest.fn();
 
 			render(
-				<ControlledUnitControl
+				<UnitControl
 					value={ 50 }
 					onChange={ onChangeSpy }
 					isPressEnterToChange
@@ -275,7 +256,7 @@ describe( 'UnitControl', () => {
 			const onBlurSpy = jest.fn();
 
 			render(
-				<ControlledUnitControl
+				<UnitControl
 					value="33%"
 					onChange={ onChangeSpy }
 					onBlur={ onBlurSpy }
@@ -304,7 +285,7 @@ describe( 'UnitControl', () => {
 			const onChangeSpy = jest.fn();
 
 			render(
-				<ControlledUnitControl
+				<UnitControl
 					value="15px"
 					onChange={ onChangeSpy }
 					isPressEnterToChange
@@ -337,7 +318,7 @@ describe( 'UnitControl', () => {
 			render(
 				<>
 					<button>Click me</button>
-					<ControlledUnitControl
+					<UnitControl
 						units={ [ { value: '%', label: '%' } ] }
 						onChange={ onChangeSpy }
 					/>
@@ -370,7 +351,7 @@ describe( 'UnitControl', () => {
 			const onUnitChangeSpy = jest.fn();
 
 			render(
-				<ControlledUnitControl
+				<UnitControl
 					value="14rem"
 					onChange={ onChangeSpy }
 					onUnitChange={ onUnitChangeSpy }
@@ -400,7 +381,7 @@ describe( 'UnitControl', () => {
 				{ value: '+', label: '+', default: 10 },
 			];
 
-			render( <ControlledUnitControl units={ units } /> );
+			render( <UnitControl units={ units } /> );
 
 			const options = getSelectOptions();
 
@@ -423,7 +404,7 @@ describe( 'UnitControl', () => {
 			];
 
 			render(
-				<ControlledUnitControl
+				<UnitControl
 					isResetValueOnUnitChange
 					units={ units }
 					onChange={ onChangeSpy }
@@ -459,7 +440,7 @@ describe( 'UnitControl', () => {
 			];
 
 			render(
-				<ControlledUnitControl
+				<UnitControl
 					isResetValueOnUnitChange={ false }
 					value={ 50 }
 					units={ units }
@@ -490,7 +471,7 @@ describe( 'UnitControl', () => {
 			const onChangeSpy = jest.fn();
 
 			render(
-				<ControlledUnitControl
+				<UnitControl
 					value="50%"
 					units={ [ { value: '%', label: '%' } ] }
 					onChange={ onChangeSpy }
@@ -550,7 +531,7 @@ describe( 'UnitControl', () => {
 				{ value: 'vmax', label: 'vmax' },
 			];
 
-			render( <ControlledUnitControl units={ units } value="5" /> );
+			render( <UnitControl units={ units } value="5" /> );
 
 			const select = getSelect();
 			await user.selectOptions( select, [ 'vmax' ] );
@@ -568,7 +549,7 @@ describe( 'UnitControl', () => {
 			const onBlurSpy = jest.fn();
 
 			render(
-				<ControlledUnitControl
+				<UnitControl
 					value="15px"
 					onUnitChange={ onUnitChangeSpy }
 					onBlur={ onBlurSpy }
@@ -593,28 +574,26 @@ describe( 'UnitControl', () => {
 
 	describe( 'Unit Parser', () => {
 		it( 'should update unit after initial render and with new unit prop', async () => {
-			const { rerender } = render(
-				<ControlledUnitControl value="10%" />
-			);
+			const { rerender } = render( <UnitControl value="10%" /> );
 
 			const select = getSelect();
 
 			expect( select.value ).toBe( '%' );
 
-			rerender( <ControlledUnitControl value="20vh" /> );
+			rerender( <UnitControl value="20vh" /> );
 
 			expect( select.value ).toBe( 'vh' );
 		} );
 
 		it( 'should fallback to default unit if parsed unit is invalid', () => {
-			render( <ControlledUnitControl value="10null" /> );
+			render( <UnitControl value="10null" /> );
 
 			expect( getSelect().value ).toBe( 'px' );
 		} );
 
 		it( 'should display valid CSS unit when not explicitly included in units list', () => {
 			render(
-				<ControlledUnitControl
+				<UnitControl
 					value="10%"
 					units={ [
 						{ value: 'px', label: 'px' },
@@ -640,7 +619,7 @@ describe( 'UnitControl', () => {
 				const onUnitChangeSpy = jest.fn();
 
 				render(
-					<ControlledUnitControl
+					<UnitControl
 						value="10%"
 						onChange={ onChangeSpy }
 						onUnitChange={ onUnitChangeSpy }
```